plugins/modem-manager/fu-mm-utils.c

```diff
@@ -12,59 +12,133 @@
 
 #include "fu-mm-utils.h"
 
+static gchar *
+find_device_bus_subsystem (GUdevDevice *device)
```

Comment:
does g_udev_device_get_parent_with_subsystem() work here instead?

Reply:
I'm not totally convinced, because `g_udev_device_get_parent_with_subsystem()` would require us to run it one by one for all the bus subsystems we may expect, and then we would need to decide which of them is closer in the hierarchy to the port GUdevDevice (as e.g. a USB device may be hanging off PCI as well). This `find_device_bus_subsystem()` will instead iterate parents until the first subsystem of the expected buses is found (e.g. on a USB device hanging off PCI we would find USB first).